Лабораторная работа №3 - Садовничий

SadovnichiyDenis opened the pull request.

SadovnichiyDenis committed:
Лабораторная работа №3 - Садовничий

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,27 @@
 package main
 
-import "fmt"
+import (
```
файлики .DS_Store добавьте в gitignore

jskonst replied:
Все еще не удалили DS_Store, а без этого не смогу заапрувить PR

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,27 @@
+}
+
+func formula(x float64) float64 {
+	var a float64 = 2.0
```
a и b тоже сделайте параметрами функции

SadovnichiyDenis committed:
Исправления

1. Добавил функции задач
2. Добавил Ds_Store  в игнор
3. Добавил константу а

jskonst commented on .gitignore:
```diff
@@ -0,0 +1,2 @@
+.DS_Store
+golang/.DS_Store
```
можно * добавить **/.DS_Store

jskonst replied:
ну и сам DS_Store надо удалить

jskonst replied:
Сам бинарный файл все еще присутствует в репозитории
https://stackoverflow.com/questions/13541615/how-to-remove-files-that-are-listed-in-the-gitignore-but-still-on-the-repositor

jskonst replied:
А почему опять файлик DsStore попал? мы ж в прошлый раз починили?

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+func TaskA(a float64) []float64 {
+	var x float64 = 1.2
+	var taskA []float64
+	for ; x < 4.2; x = x + 0.6 {
```
а почему тут не объявить x ?

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+	return taskA
+}
+
+func TaskB(a float64) []float64 {
```
сам массив значений x также передайте в виде параметров

SadovnichiyDenis committed:
Исправления 2

1. Поправил файл ГитИгнор
2. Массив х передал в виде параметров
3. Объявил х в цикле

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,36 @@
+	var masB = []float64{1.16, 1.32, 1.47, 1.65, 1, 93}
+	const a float64 = 2.0
+	fmt.Println("___Задание А___")
+	fmt.Println(TaskA(a))
```
В задачу а также нужно передать диапазон изменения х

jskonst replied:
все еще не поменялось

jskonst commented:
Почти готово - только удалить DsStore и в задачу А добавить диапазон изменения х

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,36 @@
+
+func TaskA(a float64) []float64 {
+	var taskA []float64
+	for x := 1.2; x < 4.2; x = x + 0.6 {
```
```suggestion
	for x := xn; x < xk; x += dx {
```

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,36 @@
+	fmt.Println(TaskB(a, masB))
+}
+
+func TaskA(a float64) []float64 {
```
```suggestion
func TaskA(a, xn, xk, dx float64) []float64 {
```

SadovnichiyDenis committed:
DS_Store delete

Я искренне надеюсь, что в паутине маковского терминала смог найти и удалить нужные файлы .DS_Dtore

SadovnichiyDenis committed:
Доработки по задаче А

1. Создал массив значений
2. Вставил его в задачу

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+	fmt.Println(TaskB(a, masB))
+}
+
+func TaskA(a float64, masA []float64) []float64 {
```
в задаче А передаются xn, xk, dx

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+
+func TaskB(a float64, masB []float64) []float64 {
+	var taskB []float64
+	for i := 0; i < len(masB); i++ {
```
```suggestion
	for i, item := range masB {
```

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+	return taskA
+}
+
+func TaskB(a float64, masB []float64) []float64 {
```
```suggestion
func TaskB(b float64, masB []float64) []float64 {
```

jskonst commented on golang/main.go:
```diff
@@ -1,7 +1,37 @@
+func TaskB(a float64, masB []float64) []float64 {
+	var taskB []float64
+	for i := 0; i < len(masB); i++ {
+		taskB = append(taskB, formula(masB[i], a))
```
```suggestion
		taskB = append(taskB, formula(masB[i], b))
```

SadovnichiyDenis committed:
исправления и удаление DS_Dtore

Стоит подучить функции, потому что многое не понимаю как работает.
добавил начальную, конечную и дельта х, вставил в функцию.

SadovnichiyDenis committed:
Исправил ошибки в задаче А

jskonst reviewed (changes requested):
Прошлые еще не починены

SadovnichiyDenis committed:
Задача А и ДС Сторе

Передал значения х в задачу А и удалил ДС Сторе

SadovnichiyDenis committed:
Добавил 5 задач

Первые 5 задач из Codewars

SadovnichiyDenis committed:
Выполнил все задачи по Кодеварс
